src/app/components/core/control-dispatcher/add-control-script/add-step/add-configure/add-configure.component.ts

```diff
@@ -200,6 +186,21 @@ export class AddConfigureComponent implements OnInit {
 
   }
 
+  setCategoryName(nodes) {
+    let category = nodes.find(r => r.key == this.config.category);
+    if(!category) {
+      for(let node of nodes){
+        if(node.children.length > 0) {
+          category = this.setCategoryName(node.children);
+          if(category){
+            return category;
+          }
+        }
+      }
+    }
+    return category;
```

Comment:
If the for if failed to assign then it will return `undefined` and then `setCategoryName` method will be further called recursively to check category in sibling and child nodes.